src/troute-routing/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -222,8 +222,8 @@ cpdef object compute_network_structured(
     # Check shapes
     if qlat_values.shape[0] != data_idx.shape[0]:
         raise ValueError(f"Number of rows in Qlat is incorrect: expected ({data_idx.shape[0]}), got ({qlat_values.shape[0]})")
-    if qlat_values.shape[1] < nsteps:
-        raise ValueError(f"Number of columns (timesteps) in Qlat is incorrect: expected at most ({data_idx.shape[0]}), got ({qlat_values.shape[1]}). The number of columns in Qlat must be equal to or less than the number of routing timesteps")
+    if qlat_values.shape[1] < nsteps/qts_subdivisions:
+        raise ValueError(f"Number of columns (timesteps) in Qlat is incorrect: expected at least ({data_idx.shape[1]/(nsteps/qts_subdivisions)}), got ({qlat_values.shape[1]}). The number of columns in Qlat must be at least the number of routing timesteps")
```

Comment:
qts_subdivisions in Line 225 should be equal to dt_qlat / dt, where dt_qlat and dt are lateral flow data time interval and channel routing time interval, respectively.